### ISSUE TITLE
Create a release compatible with phpunit 6.0+

### ISSUE THREAD

**Issue by TysonAndre** [brianium/paratest#226]
https://github.com/sebastianbergmann/phpunit/wiki/Release-Announcement-for-PHPUnit-6.0.0#backwards-compatibility-issues

They changed from underscores to namespaces for naming classes. I haven't checked if there are other incompatibilities.

**Comment by julianseeger** [comment]
I labeled this as a bug now. Maybe thats not very exact but as the build is failing with phpunit 6, someone has to fix it before we can create a compatible release.

**Comment by michaelbutler** [comment]
Anyone know if there is a backwards compatible polyfill to allow projects to work with both phpunit 5.X and 6.X?

**Comment by julianseeger** [comment]
As a last resort, we could create a branch for phpunit5 (to let that branch die at 02.02.2018) and create phpunit 6 compatibility in the master.
Technically it could work to use both phpunit-versions in parallel when they don't even overlap in namespaces, but getting both to work in paratest sounds like a maintenance nightmare.

**Comment by julianseeger** [comment]
Looking at the network-graph, there is at least one phpunit 6 fork from @matthewnessworthy . As far as I'm concerned, we could just switch to phpunit 6 after the next release. Composer should be able to handle that and install that release for everyone using phpunit 5 and install all further releases for everyone requiring phpunit 6.

Maybe phpunit6 support would be a reason to create some paratest v1.0.0?

**Comment by splitice** [comment]
+1

**Comment by michaelbutler** [comment]
I agree it **could** be a maintenance nightmare to have to maintain both PHPUnit 5 and 6... but let me see if I could do it.

**Comment by michaelbutler** [comment]
I was hacking around with this creating a PHPUnit 5.X polyfill if using PHPUnit 6... and indeed it started to get pretty janky. It actually did work, however I ran into the annoying process hang problem. No errors or anything, just hangs until you Ctrl-C. I'm not sure if that's directly related to what I've changed but it probably is. 

Thus I would recommend created a new branch for PHPUnit 6 support.

**Comment by redaxmedia** [comment]
What is the status of this ticket? I and maybe others cannot use your great software after switching to PHPUnit 6 ... I suggest to drop the PHPUnit 5.x support for a Paratest 0.2... !?

**Comment by julianseeger** [comment]
The version is not the problem. We could create Paratest 0.2.0, 1.0.0 or 42.0.0.
Problem is: no one yet contributed a (working) PR with PHPUnit 6 support

As PHPUnit 5 has its EOL in Feb. 2018, the current paratest version could be the last to support it and the next version would only have to support phpunit 6. if it's in the composer.json, that's good enough. So any contributions for a phpunit6-paratest are geatly appreciated.

**Comment by TysonAndre** [comment]
https://github.com/brianium/paratest/compare/master...TysonAndre:phpunit6-fix?expand=1 should be further along than the other phpunit 6 fork that was mentioned

- One command line flag for disabling saving global state was disabled

The only remaining bug seems to be that paratest isn't aware of incomplete tests

**Comment by TysonAndre** [comment]
Follow up work:

Make sure that incomplete/skipped tests are output in the below format in the junit output **of paratest** (doubt it was implemented in my PR)

```
      <testcase name="testParatestIC1" class="myTest" classname="tag_urlTest" file="/mysrc/tests/myfolder/myTest.php" line="629" assertions="0" time="0.011993">
        <skipped/>
      </testcase>
```

Make sure that the names, data provider info, etc. of skipped tests are now printed in CLI output, the same way phpunit would print those

**Comment by matthewnessworthy** [comment]
@julianseeger please see the linked PR #237 which is currently passing

**Comment by julianseeger** [comment]
🎉 🎈 🎂  good job
waiting for an ack on release 1.0.0 from @brianium 

And FYI: I closed some PRs because they had no tests and now we now every new PR is based on the phpunit6-supporting master. If anyone has the time to dig in and write tests for those PRs, they might be a quick win because the code is / was already there - just worthless without test ;)

**Comment by julianseeger** [comment]
We have an ack on 1.0.0.
And as nearly a workweek passed without major new bugs, I'm going to create release 1.0.0 in a couple of hours. Any comments / additions / critics?

**Comment by iwiznia** [comment]
Hi! I just tested this on a project. I'm noticing that tests marked as risky when running regular phpunit are treated as errors when running through paratest.
In phpunit it looks like this:
![image](https://user-images.githubusercontent.com/521248/27289203-65cd4086-5509-11e7-8315-e88734f5c7bd.png)
And no error is shown on the bottom.

While in paratest:
![image](https://user-images.githubusercontent.com/521248/27289213-6e114fee-5509-11e7-87a0-f847c49f59d9.png)
![image](https://user-images.githubusercontent.com/521248/27289235-79444772-5509-11e7-993b-e9b11fc5d475.png)
